Dockerfile

```diff
@@ -1,10 +1,14 @@
-FROM ubuntu:latest
-
-COPY ./build/arc_linux_amd64 /service/arc
+FROM golang:1.21.1
```

Comment:
The docker image when using this base image is much larger: ~302.75 MB
With ubuntu base image it is just ~50MB. You can see that here: https://hub.docker.com/r/bsvb/arc/tags

Reply:
I'm not sure we have to discuss image size. If that becomes a real problem we can optimize later. What do you think?

Reply:
I don't see the benefit in this change. We just create a larger docker image which takes longer to pull and takes more storage.

Reply:
well 
1. with previous dockerfile I wasn't able to build anything 
2. using official golang docker image is somewhat simplest way to get things done as it's recommended here
https://docs.docker.com/language/golang/build-images/
3. I'd be happy to see alternative approach in building ci that works in every environment.

Reply:
Did you run `make build_release` before building the docker container?

On my mac I had no problem to run this locally with e.g.
```
make build_release
 docker build -t arc .
```

Reply:
We can definitely keep the original way however the Dockerfile I propose is reused for tests so we avoid some overhead in it's maintenance.
What do you think?